interface/ceed-operator.c

```diff
@@ -1513,6 +1494,21 @@ int CeedOperatorApply(CeedOperator op, CeedVector in, CeedVector out, CeedReques
         CeedCall(CeedOperatorApplyAdd(op->sub_operators[i], in, out, request));
       }
     }
+  } else {
+    // Standard Operator
+    if (op->Apply) {
+      CeedCall(op->Apply(op, in, out, request));
+    } else {
+      // Zero all output vectors
+      CeedQFunction qf = op->qf;
+      for (CeedInt i = 0; i < qf->num_output_fields; i++) {
+        CeedVector vec = op->output_fields[i]->vec;
+        if (vec == CEED_VECTOR_ACTIVE) vec = out;
+        if (vec != CEED_VECTOR_NONE) CeedCall(CeedVectorSetValue(vec, 0.0));
+      }
+      // Apply
+      if (op->num_elem) CeedCall(op->ApplyAdd(op, in, out, request));
```

Comment:
Do we want to catch if we get to this point and `op->num_elem` is NULL? Seems weird that we might get this far, not actually apply the operator, but not return an error or something.

Reply:
`op->num_elem` is set to `-1` initially. Having `op->num_elem == 0` means that the operator has zero elements on the current process (think like a boundary condition operator)

Reply:
By accepting 0 elements on the current process as valid, we need a small amount of special case logic in libCEED but prevent the caller from having to wrap similar special case logic around all of their operators.

Reply:
Found it! Here's the original bugfix that added this logic: https://github.com/CEED/libCEED/pull/383

Reply:
Digging that PR from way back. Haha.

But gotcha, makes sense.